work_queue/src/network_function.py

```diff
@@ -50,18 +52,17 @@ def main():
             if event_size:
                 # receive the event itself
                 event = conn.recv(event_size)
-
                 event = json.loads(event)
-                print('event: {}'.format(event))
-
+                print('Network function: recieved event: {}'.format(event), file=sys.stderr)
                 # create a forked process for function handler
-                manager = multiprocessing.Manager()
-                response = manager.dict()
-                p = multiprocessing.Process(target=getattr(wq_worker_coprocess, function_name), args=(event, response))
-                p.start()
-                p.join()
-
-                response = json.dumps(dict(response))
+                p = os.fork()
+                if p == 0:
```

Comment:
If the fork fails, the os.read below may wait forever.  Let's do the full `if p == 0  elif p < 0 else`.